### ISSUE TITLE
pseq versus Pseq (add loop)

### ISSUE THREAD

**Issue by totalgee** [loopier/ziva#6]
https://github.com/loopier/ziva/blob/c2fb2975643aa4147f4237ed21c543015c94c1d3/Classes/sequenceablecollection.sc#L206-L208

Hi, would you consider making your `pseq` helper have the default number of repeats be 1, just like the library class `Pseq` itself? For infinite repeats over collections, I would add a new `loop` method:

```
+ SequenceableCollection {

	pseq { arg repeats=1, offset=0;
		^Pseq(this, repeats, offset)
	}

	loop {
		^Pseq(this, inf)
	}

...
```

This would require changes to the places in Ziva where `something.pseq` is called (without any arguments) to be replaced by `something.pseq(inf)` or `something.loop`.

The reason I ask is a bit selfish, because [Bacalao](https://github.com/totalgee/bacalao) defines its own `pseq` this way (consistent with the library class), so there is a conflict between Ziva and Bacalao, and people with both live coding libraries installed will get incorrect behaviour in one place or the other (playing just once or playing forever, when the other case was expected).

**Comment by loopier** [comment]
I've been considering options other than `pseq` -- and other pattern methods for arrays. The leading `p` kind of annoys me, although it serves to differentiate some `Pattern` methods from their `Array` equivalents. I first thought your `loop` idea was cool but, thinking again, I realized it doesn't describe the pattern. Any pattern could loop. I'd rather prefer something like `seq`, but removing the `p` doesn't work for all patterns. For example `prand` could not be just  `rand` because there's already an `Array` method called `rand`.

Did you choose to keep the library class default only to be consistent with it? I can understand that, but I've hacked so many things so far that Živa is anything but consistent with SC :/

I'd like to find a solution to this. I'm very open to more ideas.

> This would require changes to the places in Ziva where `something.pseq` is called (without any arguments) to be replaced by

My self from the past took care of this (I think I knew something like  this might rise up) and never used `.pseq` in Ziva's own code, only `Pseq(...)`.

**Comment by totalgee** [comment]
> My self from the past took care of this (I think I knew something like this might rise up) and never used .pseq in Ziva's own code, only Pseq(...).

You actually do use it in several places. The problematic one for me is here (this prevents Ziva `play` lines from looping):

https://github.com/loopier/ziva/blob/master/Classes/pattern.sc#L151

and probably also:

https://github.com/loopier/ziva/blob/master/Classes/ndef.sc#L18

These could just be changed to `args = blah.pseq(inf)` and that would work fine for me.

**Comment by loopier** [comment]
Ah, that's was my present self, not my past self. I changed them to `Pseq(.../ inf)`. See if it works now.

**Comment by totalgee** [comment]
Fixed by eb8038e (and 9e92132).
